dataprofiler/tests/profilers/test_categorical_column_profile.py

```diff
@@ -49,6 +49,43 @@ def test_correct_categorical_model_string(self):
         }
         self.assertCountEqual(categories, profile.categories)
 
+    def test_stop_condition_is_met_initially(self):
+        dataset = self.aws_dataset["host"].dropna()
+        profile = CategoricalColumn(dataset.name)
+        profile.max_sample_size_to_check_stop_condition = 200
+        profile.stop_condition_unique_value_ratio = 0
+        profile.update(dataset)
+
+        self.assertTrue(profile._stop_condition_is_met)
+        self.assertEqual(profile.categories, [])
+        self.assertEqual(profile.unique_count, 0.0)
+        self.assertEqual(profile.unique_ratio, 0.0)
```

Comment:
Since we are computing the categories and merging already, we should use that information to update these values.
Hence, since this is a new dataset, I would expect the `unique_ratio` and `unique_count` == to that of the dataset.
In the toy example above: `# 20 samples, unique ratio = 0.10, unique_count = 4, sample_size = 40`